cirq-core/cirq/ops/moment.py

```diff
@@ -327,6 +332,58 @@ def transform_qubits(
         """
         return self.__class__(op.transform_qubits(qubit_map) for op in self.operations)
 
+    def expand_to(self, qubits: Iterable['cirq.Qid']) -> 'cirq.Moment':
+        """Returns self expanded to given superset of qubits by making identities explicit."""
+        if not self.qubits.issubset(qubits):
+            raise ValueError(f'{qubits} is not a superset of {self.qubits}')
+
+        operations = list(self.operations)
+        for q in set(qubits) - self.qubits:
+            operations.append(ops.I(q))
+        return Moment(*operations)
+
+    def _kraus_(self) -> Sequence[np.ndarray]:
+        """Returns Kraus representation of self."""
```

Comment:
Can you add some more details to the docstring? For example, a mention that the method performs parallel concatenation of kraus operators of the individual operations ? 

Also, do you think it would be useful to have methods in `cirq/qis/channels` for serial and parallel concatenation of mixtures and kraus operators, which can be useful general primitives?

The parallel concatenation method can be used here and the serial concatenation method can be used in the kraus/mixture protocol methods while computing operators of an operation using the operators of it's decomposed operations.

Reply:
Re docstring: Done.

Re composition: Composing Kraus representations (in serial or in parallel) is cumbersome because it results in a representation with a very large number of Kraus operators. We should avoid it whenever possible. For serial composition we can always(*) avoid it by using superoperator representation instead. For parallel composition all our channel representations are cumbersome (e.g. superoperator becomes a very large matrix). However, we can still describe channels using circuits which avoids the issue. Thus, we should not need to compute serial composition of Kraus representations (else we're doing it wrong) and we should only compute parallel composition in the context of a circuit moment. For this reason, I'd rather we don't implement serial composition of Kraus representations at all and keep the code for parallel composition in `Moment._kraus_`.

---

(*) It's interesting how parallel (or "spatial") composition preserves all the details of the composed processes preventing us from compressing it while serial (or "temporal") composition loses some information about what's going on enabling a compressed description. This observation has inspired [a lot of cool ideas](https://arxiv.org/abs/gr-qc/0608043).

Reply:
BTW: Note that #4486 concerns serial composition only. Thus, we will be able to solve the issues encountered there once this PR and #4545 are merged since then we'll have access to fully supported superoperator representation.

Reply:
Makes sense, thanks for the explanation!